aderyn_evaluator/examples/detector_description.rs

```diff
@@ -0,0 +1,109 @@
+presented to it, we make the detectors battle again where we flip the position of the
+descriptions and re-ask the same question. (That's why we have `q * 2`)
+
+The above trick is shamelessly borrowed from the paper -
+`Peer Rank and Discussion Improve Large Language Model based Evaluations`
+
+Published URL: https://arxiv.org/abs/2307.02762
+
+(The authors tried to evaluate various LLMs outputs, but here we do that to
+provided detectors' descriptions)
+
+The default values for initializing elo ratings are picked from
+https://colab.research.google.com/drive/1lAQ9cKVErXI1rEYq7hTKNaCQ5Q8TzrI5?usp=sharing#scrollTo=QLGc6DwxyvQc
+
+More on how Elo Rating works can be found at (Wiki link) -
+https://en.wikipedia.org/wiki/Elo_rating_system
+
+*/
+
+use std::{collections::HashMap, error::Error};
+
+use aderyn_driver::detector::Detector;
+use aderyn_evaluator::llm::detector_description;
+
+#[derive(Default)]
+pub struct DummyDetectorA;
+
+#[derive(Default)]
+pub struct DummyDetectorB;
+
+#[derive(Default)]
+pub struct DummyDetectorC;
+
+impl Detector for DummyDetectorA {
+    // Below text has been copy-pasted from code4rena automated findings.
+    // https://github.com/code-423n4/2023-12-initcapital/blob/main/4naly3er-report.md
+    fn description(&self) -> String {
+        String::from(
+            "If not cached, the solidity compiler will always read the length of the array during each iteration. That is, if it is a storage array, this is an extra sload operation (100 additional extra gas for each iteration except for the first) and if it is a memory array, this is an extra mload operation (3 additional gas for each iteration except for the first)."
+        )
+    }
+}
+
+impl Detector for DummyDetectorB {
+    fn description(&self) -> String {
+        String::from("Array length will be read from storage in every loop and it can be expensive")
+    }
+}
+
+impl Detector for DummyDetectorC {
+    fn description(&self) -> String {
+        String::from("Array length not cached in loop.")
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // The 3 detectors will be our contestants
+
+    let mut contestants: HashMap<String, Box<dyn Detector>> = HashMap::new();
+
+    contestants.insert("det-A".to_string(), Box::<DummyDetectorA>::default());
+    contestants.insert("det-B".to_string(), Box::<DummyDetectorB>::default());
+    contestants.insert("det-C".to_string(), Box::<DummyDetectorC>::default());
+
+    // How we want the descriptions of our "ideal" detector of this type
+    // List out what makes us satisifed about the description
+
+    let satisfiers = vec![
+        // These are the "questions" referenced above
+        "It informs the user about specific details such extra gas cost that can be saved"
+            .to_owned(),
+        "It explains in detail what has gone wrong".to_owned(),
+    ];
+
+    detector_description::run(contestants, satisfiers).await?;
```

Comment:
We just have to configure the contestants and satisifers. Rest will take of itself